scripts/cddl/utils.js

```diff
@@ -2,35 +2,37 @@
  * Helper function to iterate through the parsed spec document and
  * filter out all `<pre>` nodes with class `cddl`, e.g.
  * ```
- * <pre class="cddl remote-cddl">...</pre>
- * <pre class="cddl local-cddl">...</pre>
+ * <pre class='cddl remote-cddl'>...</pre>
+ * <pre class='cddl local-cddl'>...</pre>
  * ```
- * 
+ *
  * @param {[Document] | DefaultTreeNode[]} nodes tree nodes
- * @return {
- *   {
- *     nodeClass: string,
- *     content: string[]
- *   }[]
- * } a list of pre texts
+ * @return { 'cddlName': [content: string[]] }
+ *     A list of pre texts grouped by CDDL name (eg. local).
  */
-function getCDDLNodes (nodes) {
-    return nodes.map((node) => {
-        if (node.nodeName === 'pre') {
-            const nodeClass = node.attrs.find((attr) => attr.name === 'class')
-            const content = node.childNodes.map((child) => child.value).join('')
+function getCDDLNodes(nodes) {
+  const data = nodes.map((node) => {
+    if (node.nodeName === 'pre') {
+      const nodeClass = node.attrs.find((attr) => attr.name === 'class')
+      if (nodeClass && nodeClass.value.includes('cddl')) {
```

Comment:
All the string manipulation feels like it's being far too clever. Since there are only two cases `cddl-remote` and `cddl-local` I'd just check for those explicitly, and return the nodes for each as a object like `{local: [], remote: []}`, rather than a flat list that later has to be processed again.

Reply:
Indeed. Changing that would make the code all the way easier to understand and faster. I'll update that.